src/components/NewGalaxyModal.js

```diff
@@ -39,32 +50,29 @@ export const NewGalaxyModal = withModalDrawer({ width: 675 })(class NewGalaxyMod
     //todo work with leo team
   }
 
+  applyGalaxyChanges() {
+    const { viewMode } = this.state
+    const { app: oldApp } = this.getOldEnvironmentConfig()
+    return Utils.switchCase(viewMode,
+      ['deleteWarn', () => Ajax().Apps.app(oldApp.googleProject, oldApp.appName).delete()],
+      ['createWarn', () => this.createGalaxy()],
+      [Utils.DEFAULT, () => !!oldApp ? this.setState({ viewMode: 'deleteWarn' }) : this.setState({ viewMode: 'createWarn' })]
```

Comment:
It may be worth pulling out the state changes into its own button component (similar to the new NewClusterModal), but this seems fine for now.

Reply:
I feel like we should be consistent with the create and delete and either have both ajax calls right in here or in their own functions. I think my preference is to put them in their own functions for quick readability and possibly error handling but could be swayed